### PR TITLE
Flexible selectors

### PR DESCRIPTION
**Pull request by torotil**
Remove postcode as hard-coded selector. Standardize handling of custom-datasets / postcode datasets.

### CODE REVIEW

File: campaignion_email_to_target/src/Action.php [hunk 1]
```diff
@@ -100,45 +130,51 @@ public function testLink($title, $query = [], $options = []) {
...
     $no_target_message = NULL;
-    foreach ($constituencies as $constituency) {
+
+    $selector = $this->buildSelector($submission_o);
+    $contacts = $this->api->getTargets($this->options['dataset_name'], $selector);
+
+    foreach ($contacts as $target) {
+      $target += ['constituency' => []];
+      $constituency = $target['constituency'];
+      unset($target['constituency']);
```
Comment (m-strasser) [inline campaignion_email_to_target/src/Action.php:144]: I don't really understand what's happening here, tbh ;)

Comment (torotil) [reply campaignion_email_to_target/src/Action.php:144]: Previously the contacts were grouped by constituency (in the API response `/v2/…/postcode/…` endpoint):

```python
[
  {'name': 'constituency1', contacts: [ {}, {}, {} ]},
  {'name': 'constituency2', contacts: [ {}, {}, {} ]},
]
```

With the new endpoint (`/v2/…/select?…`) it’s

```python
[
  {'name': 'contact1', constituency: {}},
  {'name': 'contact without constituency'},
]
```
`$contacts` in the code above is the API-response. So `$target` becomes the contact without the constituency part. And `$constituency == $target['constituency']` or an empty array if that doesn’t exist.

Comment (torotil) [reply campaignion_email_to_target/src/Action.php:144]:
```php
// This …
$target += ['constituency' => []];
// is equivalent to
if (!array_key_exists('constituency', $target)) {
  $target['constituency'] = [];
}

Comment (m-strasser) [reply campaignion_email_to_target/src/Action.php:144]: Oh okay, that explains it :)

**Review by m-strasser — APPROVED** [review]

**Pull request merged**